cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsInstanceConnector.java

```diff
@@ -160,10 +167,16 @@ private List<CloudVmInstanceStatus> setCloudVmInstanceStatuses(AuthenticatedCont
             Long timeboundInMs) {
         AmazonEc2Client amazonEC2Client = new AuthenticatedContextView(ac).getAmazonEC2Client();
         try {
-            Collection<String> instances = instanceIdsWhichAreNotInCorrectState(vms, amazonEC2Client, status, completedStatuses);
+            ImmutablePair<Collection<String>, Collection<String>> instancesPair =
```

Comment:
General question - @lajosrodek , @sodre90  - do we want this change to affect operations which are not related to fast scaling?
I would be more comfortable enabling it for fast scaling only initially - since this is a tough-to-test reproduce scenario.
(If I'm reading the code properly - an instance in PENDING state is going to break a bunch of operations with the current patch)